docs/preview/yamlgettingstarted-templates.md

```diff
@@ -0,0 +1,295 @@
+# YAML getting started - Templates (not available, for discussion only)
+
+## Example: Simple input
+
+The following example illustrates two things:
+
+1. Front-matter is used to define the default values for inputs.
+
+2. Inputs are expanded using template expressions. A value that starts and ends
+   with `${{ }}` indicates a template expression.
+
+Template:
+
+```yaml
+---
+# Default values
+inputs:
+  solution: '**/*.sln'
+---
+steps:
+- task: msbuild@1
+  inputs:
+    solution: ${{ inputs.solution }}
+- task: vstest@2
+  inputs:
+    solution: ${{ inputs.solution }}
+```
+
+Consumer:
+
+```yaml
+steps:
+- template: steps/msbuild.yml
+  inputs:
+    solution: my.sln
+```
+
+## Example: Insert into an array
+
+Template:
+
+```yaml
+---
+# Default values
+inputs:
+  preBuild: []
+  postBuild: []
+---
+phases:
+- phase: build
+  steps:
+  - ${{ inputs.preBuild }}
+  - task: msbuild@1
+  - task: vstest@2
+  - ${{ inputs.postBuild }}
+```
+
+Consumer:
+
+```yaml
+phases:
+- template: phases/build.yml
```

Comment:
I think this is a really solid design that covers a lot of ground.  Wrapping steps is what we do quite a bit for .NET Core.